### ISSUE TITLE
Reusing environements when extending MPSComoving

### ISSUE THREAD

**Issue by HumpyBlumpy** [maartenvd/MPSKit.jl#39]
Hello again!

I have implemented window expansion during time evolution of an MPSComoving (along the lines of arxiv:1207.0862,arXiv: 1207.0691v2). Every timestep, I am comparing the entanglement entropy at some bond near the edge to the ground state value and if the difference is smaller than some threshold I append a ground state tensor into the window. 
It seems to work well but I am looking into ways to speed it up.
In particular, right now every time I extend the MPS window, I have to recalculate the environments by calling `environments(mpco,th,lenvs=gs_envs,renvs=gs_envs)`.
I was wondering whether it is possible to reuse the environments prior to the expansion, instead of fully recalculating them?

Thanks!

**Comment by maartenvd** [comment]
yes, I can extend the code in such a way that you could re-use the previous environments. If you give me the code you use for extending the mps, then I will change it to also return an extended environment object (though does it really take a lot of time to recalculate these environments?). As an alternative source of speedup, I think I now converge the exponentiation step in TDVP way too far (tol there should be on the order of dt^2, instead of the default 1e-12).

I always wanted to implement these moving algorithms (hence the name mpscomoving), but there were some oddities that need to be worked out. In particular, mpscomoving always indexes the first tensor as sitting on the physical site [1]. So if you insert a new tensor at the left, every index gets mapped to [i] -> [i+1]. 

I find this annoying, and I think it wouldn't be that much work to update mpscomoving so that it physically describes a general patch of tensors (from site s_1 to site s_end), and that indexing directly corresponds to the actual physical location.

**Comment by maartenvd** [comment]
so for your specific example (a one site unitcell) I would use the following code:

https://gist.github.com/maartenvd/2a37952d8f634228178fc8712fb86822

but as I mentioned, I'm not entirely happy with the current mpscomoving design, and I hope to improve on it in the future (along with exporting 'proper' leftextend/rightextend code, that deals with bigger unit cells).

**Comment by HumpyBlumpy** [comment]
Great! thanks!
